als-server/shared/src/main/scala/org/mulesoft/als/server/modules/configuration/WorkspaceConfigurationManager.scala

```diff
@@ -41,19 +41,20 @@ class GetWorkspaceConfigurationRequestHandler(val workspaceManager: WorkspaceMan
   override protected def task(params: GetWorkspaceConfigurationParams): Future[GetWorkspaceConfigurationResult] =
     workspaceManager
       .getWorkspace(params.textDocument.uri)
-      .map(workspace => {
+      .flatMap(w => w.getCurrentConfiguration.map(c => (w, c)))
+      .map(t =>
         GetWorkspaceConfigurationResult(
-          workspace.folderUri,
-          workspace.workspaceConfiguration
```

Comment:
Should `workspaceConfiguration` now be private to prevent unwanted access? We should only enable `getCurrentConfiguration` to return it.